astropy/io/fits/connect.py

```diff
@@ -141,20 +146,28 @@ def read_table_fits(input, hdu=None, astropy_native=False):
     # Convert to an astropy.table.Table object
     # Note: in future, it may make more sense to do this column-by-column,
     # rather than via the structured array.
-    t = Table(table.data, masked=masked)
+    t = Table(masked=masked)
+
+    # In the loop below we access the data using data[col.name] rather than
+    # col.array to make sure that the data is scaled correctly if needed.
+    data = table.data
+
+    columns = []
+    for col in data.columns:
+        columns.append(Column(data=data[col.name], name=col.name, copy=False))
+    t.add_columns(columns, copy=False)
```

Comment:
Can we just do
```
t = Table(columns, masked=False, copy=False)
```
here and avoid the separate table creation above?

Reply:
I had issues doing this - I'll try again though to be sure.

Reply:
Ah actually it's fine now I'm instantiating the columns manually. Fixed.